### ISSUE TITLE
Foundation variant is failing lighthouse accessibility audit

### ISSUE THREAD

**Issue by G-Rath** [ackama/rails-template#212]
```
Failures:

  1) Homepage behaves like an accessible page passes a Lighthouse accessibility audit
     Failure/Error: expect(page).to pass_lighthouse_audit(:accessibility)

       expected http://127.0.0.1:37619/ to pass Lighthouse accessibility audit
       with a minimum score of 100

     [Screenshot Image]: /home/runner/work/rails-template/rails-template/template-test/dummy/foundation-layout-test-app/tmp/screenshots/failures_r_spec_example_groups_homepage_behaves_like_an_accessible_page_passes_a_lighthouse_accessibility_audit_931.png


     Shared Example Group: "an accessible page" called from ./spec/system/home_feature_spec.rb:12
     # ./spec/support/shared_examples/an_accessible_page.rb:7:in `block (2 levels) in <top (required)>'

Finished in 42.62 seconds (files took 6.06 seconds to load)
13 examples, 1 failure
```

**Comment by fractalcactus** [comment]
intriguing, running the audit in chrome passes 
<img width="1432" alt="image" src="https://user-images.githubusercontent.com/8527715/106052210-32a5c480-614e-11eb-986e-f4c590b82681.png">

**Comment by G-Rath** [comment]
Keep in mind the error message doesn't say what the audit score is, so it's quite possible that it's erroring for some other reason, and that the matcher library just defaults to a score of i.e 0, causing it to fail.

(that itself is probably a bug in the matcher library, which we should look to address at some point too)

**Comment by fractalcactus** [comment]
interestingly, with a score of 50, `expect(page).to pass_lighthouse_audit(:accessibility, score: 50)`, the spec passes

it would be super helpful to get the output score, so going to work on that first:
https://github.com/ackama/lighthouse-matchers/issues/14

**Comment by fractalcactus** [comment]
<img width="727" alt="image" src="https://user-images.githubusercontent.com/8527715/106085086-6d2a5400-6184-11eb-90cd-654ab3e8f7cd.png">

**Comment by fractalcactus** [comment]
opinions on this conundrum would be appreciated!
:star: :star: :star: :star: :star: :star: :star: :star: :star: :star: :star: :star: :star:
the Foundation variant is failing the lighthouse accessibility audit.
this is caused by the code in  variants/frontend-foundation-layout/app/views/application/header/_navigation.html.erb

Foundation for Sites is a fully-accessible framework. It will generate the correct structure of roles for the HTML.
e.g this code. . .
<ul class="vertical menu accordion-menu" data-accordion-menu>
  <li>
    <a href="#">Item 1</a>
    <ul class="menu vertical nested">
      <li><a href="#">Item 1A</a></li>
      <li><a href="#">Item 1B</a></li>
    </ul>
  </li>
  <li><a href="#">Item 2</a></li>
</ul>
will generate this code . . .
<ul class="vertical menu accordion-menu" data-accordion-menu="i5dgjg-accordion-menu" style="max-width: 250px" role="tree" aria-multiselectable="true">
  <li role="treeitem" class="is-accordion-submenu-parent" aria-controls="whmj8k-acc-menu" aria-expanded="false" id="y4v9co-acc-menu-link">
    <a href="#" role="menuitem">Item 1</a>
    <ul class="menu vertical nested submenu is-accordion-submenu" data-submenu="" role="group" style="display: none;" aria-labelledby="y4v9co-acc-menu-link" aria-hidden="true" id="whmj8k-acc-menu">
      <li role="treeitem" class="is-accordion-submenu-parent is-submenu-item is-accordion-submenu-item" aria-controls="m4qji0-acc-menu" aria-expanded="false" id="b6x7k3-acc-menu-link">
        <a href="#" role="menuitem">Item 1A</a>
        <ul class="menu vertical nested submenu is-accordion-submenu" data-submenu="" role="group" style="display: none;" aria-labelledby="b6x7k3-acc-menu-link" aria-hidden="true" id="m4qji0-acc-menu">
          <li role="treeitem" class="is-submenu-item is-accordion-submenu-item"><a href="#" role="menuitem">Item 1Ai</a></li>
          <li role="treeitem" class="is-submenu-item is-accordion-submenu-item"><a href="#" role="menuitem">Item 1Aii</a></li>
          <li role="treeitem" class="is-submenu-item is-accordion-submenu-item"><a href="#" role="menuitem">Item 1Aiii</a></li>
        </ul>
      </li>
      <li role="treeitem" class="is-submenu-item is-accordion-submenu-item"><a href="#" role="menuitem">Item 1B</a></li>
      <li role="treeitem" class="is-submenu-item is-accordion-submenu-item"><a href="#" role="menuitem">Item 1C</a></li>
    </ul>
  </li>
  <li role="treeitem" class="is-accordion-submenu-parent" aria-controls="agncs9-acc-menu" aria-expanded="false" id="hm42my-acc-menu-link">
    <a href="#" role="menuitem">Item 2</a>
    <ul class="menu vertical nested submenu is-accordion-submenu" data-submenu="" role="group" style="display: none;" aria-labelledby="hm42my-acc-menu-link" aria-hidden="true" id="agncs9-acc-menu">
      <li role="treeitem" class="is-submenu-item is-accordion-submenu-item"><a href="#" role="menuitem">Item 2A</a></li>
      <li role="treeitem" class="is-submenu-item is-accordion-submenu-item"><a href="#" role="menuitem">Item 2B</a></li>
    </ul>
  </li>
  <li role="treeitem"><a href="#" role="menuitem">Item 3</a></li>
</ul>
in the code snippet directly above, note the generation of ARIA roles, e.g the top level element has been assigned role="tree"
These area roles are nested correctly e.g menuitem is a child of a menu

Our issue is that _navigation.html.erb does not generate the ARIA roles correctly. We have customised foundation, we mix it in using sass to build our own class structure out of the foundation components. Our customisations have messed up the the automatic generation of correctly nested ARIA roles.

Here’s an example of the generated HTML in the template:
this dropdown generates a parent with role="treeitem", and contains a child called role="menuitem" . This ARIA nesting is incorrect, an element with role="menuitem" should always be a child of an element with role="menu"

**Comment by fractalcactus** [comment]
no consensus reached yet about whether to rip out the nav, rebuild in foundation, or hold off while discussing other front end frameworks

**Comment by mischa-s** [comment]
@fractalcactus - I think short term if you want to just start merging code in, I'd be happy for you to comment out the nav code which is failing until a proper decision is reached on what to do about it

**Comment by mischa-s** [comment]
My general sense is it would be good to refactor the nav code to pass the accessibility audit, but not sure if that's a general feeling

**Comment by mischa-s** [comment]
To maybe be a bit clearer, my sense of the 3 options is:
1. rip out the nav - short solution to unblock merging
2. rebuild in foundation nav - probably proper solution
3. hold off while discussing other front end frameworks unlikely to happen within a 6 month timeframe

**Comment by G-Rath** [comment]
This seems to have come right as CI is passing, so I'm going to close this issue.

**Comment by G-Rath** [comment]
Re-opening because it seems it's just really flakey and so still a problem.
